squash-the-creeps/rust/cheatseet.txt

```diff
@@ -0,0 +1,98 @@
+There are some guidelines you can rely on (which you probably already know):
+For Godot-native methods
+There's no dynamic typing, so you need to cast your object to your desired type before attempting to call methods of that type on it.
+
+Const methods
+Calling can be done with either <&MyClass>::base().method() or <&Gd<MyClass>>::method().
+Both the godot documentation and the gdext-rust bindings specify which are const and which aren't, the gdext bindings specify that through method signatures (eg: fn get_position(&Node2D) vs fn add_child(&mut Node)).
+
+Non-const methods
+Calling can be done with either <&mut MyClass>::base_mut().method() or <&mut Gd<MyClass>::method().
```

Comment:
Calling _what_ (built-in, inherited methods)? 

I would use snippets instead of mental shortcuts there (and further on), for example:

```rs
let mut obj: Gd<MyClass> = MyClass.new_alloc();

// To call method with &self receiver 
obj.bind().method();

// To call method with &mut self receiver 
obj.bind_mut().method();
```

`<Type>` has some specific meaning in rust, so it is misleading.